### ISSUE TITLE
[export] “Multiple exports” false positive on merged class and namespace

### ISSUE THREAD

**Issue by andersk** [import-js/eslint-plugin-import#1964]
```ts
// foo.ts
export class Foo {
  bar: string;
  constructor(options: Foo.Options) {
    this.bar = options.bar;
  }
}

export namespace Foo {
  export interface Options {
    bar: string;
  }
}
```

```json
{
  "parser": "@typescript-eslint/parser",
  "plugins": ["import"],
  "rules": {
    "import/export": "error"
  }
}
```

```
/tmp/test/foo.ts
  1:14  error  Multiple exports of name 'Foo'  import/export
  8:18  error  Multiple exports of name 'Foo'  import/export

✖ 2 problems (2 errors, 0 warnings)
```

TypeScript [namespace merging](https://www.typescriptlang.org/docs/handbook/declaration-merging.html#merging-namespaces-with-classes-functions-and-enums) should not be an error. This is similar to #1549, but that issue seems to be specific to default exports.

**Comment by smac89** [comment]
For me, the problem comes from doing something like:

```ts
export enum Sex {
    MALE,
    FEMALE,
}

export namespace Sex {
    export function parse(sex: string): Sex {
        switch (sex.toLowerCase()) {
            case 'male':
                return Sex.MALE;
            case 'female':
                return Sex.FEMALE
            default:
                return null;
        }
    }
}
```

The error is on the `parse` function inside the namespace even though it is being used in some other parts of the code.

**Comment by ljharb** [comment]
Setting aside that your example is biologically inaccurate, thanks for the additional reproduction code. I'd be happy to review a PR if someone has a solution.

**Comment by ljharb** [comment]
@httpete if you're on the latest version of the plugin and still having problems, please file a new issue.
